repos/spack_repo/builtin/packages/python/package.py

```diff
@@ -173,6 +173,7 @@ class Python(Package):
     variant("tix", default=False, description="Build Tix module", when="+tkinter")
     variant("crypt", default=True, description="Build crypt module", when="@:3.12 platform=linux")
     variant("crypt", default=True, description="Build crypt module", when="@:3.12 platform=darwin")
+    variant("gil", default=True, description="Build with Global Interpreter Lock", when="@3.13:")
```

Comment:
What about encoding that the GIL is always enabled for earlier python, and this is different from a new feature? 
```suggestion
    variant("gil", default=True, description="Build with Global Interpreter Lock")
    conflicts("~gil", when="@:3.12", msg="GIL cannot be disabled before 3.13")
```